In [3]:
import pandas as pd
dataset=pd.read_csv("E:/LetsUpgrade/Day_24/train.csv")
from sklearn import preprocessing
from sklearn.model_selection import train_test_split 
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix
from sklearn import svm

## Preprocessing

In [4]:
le=preprocessing.LabelEncoder()
le.fit(dataset["Sex"])
dataset["Sex"]=le.transform(dataset["Sex"])
le.fit(dataset["Embarked"])
dataset["Embarked"]=le.transform(dataset["Embarked"])

## Setting Independent Variables Dataset

In [5]:
X=dataset.drop(["Survived","PassengerId","Name","Ticket","Cabin"],axis=1)

## Implementing Function for SVM and classifying data using HyperPlane

In [7]:
def func(dataset,y1):
    y=dataset[y1]
    X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)
    clf=svm.SVC(gamma=0.01,C=100)
    clf.fit(X_train,y_train)
    y_pred=clf.predict(X_test)
    print("Accuracy :",accuracy_score(y_test,y_pred,normalize=True))
    print("Confusion matrix : \n",confusion_matrix(y_test,y_pred))
    
DVList=["Pclass","Survived","Sex","SibSp","Parch","Embarked"]
for i in DVList:
    print("\nDV is",i)
    func(dataset,i)


DV is Pclass
Accuracy : 0.9700374531835206
Confusion matrix : 
 [[ 67   2   1]
 [  2  46   1]
 [  0   2 146]]

DV is Survived
Accuracy : 0.7415730337078652
Confusion matrix : 
 [[124  33]
 [ 36  74]]

DV is Sex
Accuracy : 0.9213483146067416
Confusion matrix : 
 [[ 90   8]
 [ 13 156]]

DV is SibSp
Accuracy : 0.8651685393258427
Confusion matrix : 
 [[168  10   4   0   0   0   0]
 [ 14  48   3   0   0   0   0]
 [  2   2   4   0   0   0   0]
 [  0   0   1   6   0   0   0]
 [  0   0   0   0   2   0   0]
 [  0   0   0   0   0   1   0]
 [  0   0   0   0   0   0   2]]

DV is Parch
Accuracy : 0.8838951310861424
Confusion matrix : 
 [[194   1   3   0   0]
 [ 14  25   1   0   0]
 [  6   4  17   0   0]
 [  0   0   1   0   0]
 [  0   0   1   0   0]]

DV is Embarked
Accuracy : 0.9662921348314607
Confusion matrix : 
 [[ 46   0   3]
 [  0  22   0]
 [  5   1 190]]


## Result

**Inference:**
* When DV is Passenger class, accuracy is 97%
* When DV is Survived, accuracy is 74.15%
* When DV is Gender, accuracy is 92.13%
* When DV is Sibling, accuracy is 86.51% 
* When DV is Parent-child, accuracy is 88.38%
* When DV is Embarked, accuracy is 96.62% 

_Hence, the data classified using Hyper Plane, gives the best result with variables Passenger class, Gender and Embarked._